### PR TITLE
Set up pybind11_json target's public headers instead of manual installation

### PR DESCRIPTION
**Pull request by SylvainCorlay**

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
cc @martinRenou

**Comment by martinRenou** [comment]
Good to be merged?

**Comment by SylvainCorlay** [comment]
I think so. Maybe @JohanMabille wants to comment.

**Comment by SylvainCorlay** [comment]
There is an error in the xtl counterpart pr. looking into it.

File: CMakeLists.txt [hunk 1]
```diff
@@ -45,6 +45,10 @@ target_include_directories(${PROJECT_NAME}
                            $<BUILD_INTERFACE:${PYBIND_JSON_INCLUDE_DIR}>
                            $<INSTALL_INTERFACE:include>)
 
+set_target_properties(${PROJECT_NAME}
+        PROPERTIES
+        PUBLIC_HEADER "${PYBIND11_JSON_HEADERS}")
```
Comment (JohanMabille) [inline CMakeLists.txt]: Isn't it redundant with the `PUBLIC_HEADER` argument of the `install` command?

Comment (SylvainCorlay) [reply CMakeLists.txt]: My understanding is that `set_target_properties` defines the public headers, and the `install` command is given the destination for public headers.

**Comment by SylvainCorlay** [comment]
The xtl CI is complaining the `PUBLIC_HEADER` is not a whitelisted property for INTERFACE libraries.

**Comment by SylvainCorlay** [comment]
OK, that one is ready!

As per the [cmake 3.15 release notes](https://cmake.org/cmake/help/v3.15/release/3.15.html#properties
), the PUBLIC_HEADER property was only hadded for INTERFACE libraries with 3.15.

So I only make that change conditionaly to cmake 3.15 being in use.

**Pull request merged**